pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -173,7 +173,16 @@ func (ps *PipelineSpec) Validate(ctx context.Context) *apis.FieldError {
 
 	// Validate the pipeline task graph
 	if err := validateGraph(ps.Tasks); err != nil {
-		return apis.ErrInvalidValue(err.Error(), "spec.tasks")
+		ferr := apis.ErrInvalidValue(err.Error(), "spec.tasks")
+		ferr.Details = "Invalid Graph"
+		return ferr
+	}
+
+	// Validate the pipeline finally graph
+	if err := validateGraph(ps.Finally); err != nil {
```

Comment:
This check should not be added here. `finally` section is a list of tasks not graph from users perspective. Tasks are validated to not include `runAfter` [here](https://github.com/tektoncd/pipeline/blob/master/pkg/apis/pipeline/v1beta1/pipeline_validation.go#L447). This check will result in confusion. I would rather have the checks be done explicitly for `runAfter` the way its implemented.

Reply:
The only reason I added the check here was to be consistent with the current behaviour. We have a unit test right now that enforces emitting an `InvalidDagError` to the user in case the finally DAG is not valid, and the only way to keep that with the new code org was to add it here.

I'd be happy to drop this and drop the unit test and just bubble up any validation error as it's reported by the dag module directly.

Reply:
yes please, will add it back when we change `finally` to be graph instead of list, thanks 🙏